Mage.Sets/src/mage/cards/m/MrHousePresidentAndCEO.java

```diff
@@ -0,0 +1,159 @@
+        this.addAbility(new MrHousePresidentAndCEOTriggeredAbility());
+
+        // {4}, {T}: Roll a six-sided die plus an additional six-sided die for each mana from Treasures spent to activate this ability.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new MrHousePresidentAndCEODieRollEffect(), new ManaCostsImpl<>("{4}"));
+        ability.addCost(new TapSourceCost());
+        this.addAbility(ability);
+    }
+
+    private MrHousePresidentAndCEO(final MrHousePresidentAndCEO card) {
+        super(card);
+    }
+
+    @Override
+    public MrHousePresidentAndCEO copy() {
+        return new MrHousePresidentAndCEO(this);
+    }
+}
+
+// Based on As Luck Would Have It
+class MrHousePresidentAndCEOTriggeredAbility extends TriggeredAbilityImpl {
+
+    public MrHousePresidentAndCEOTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new MrHousePresidentAndCEOTokenEffect(), false);
+    }
+
+    private MrHousePresidentAndCEOTriggeredAbility(final MrHousePresidentAndCEOTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public MrHousePresidentAndCEOTriggeredAbility copy() {
+        return new MrHousePresidentAndCEOTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DIE_ROLLED;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        DieRolledEvent drEvent = (DieRolledEvent) event;
+        if (this.isControlledBy(event.getPlayerId()) && drEvent.getRollDieType() == RollDieType.NUMERICAL) {
+            // looks for "result" instead "natural result"
+            int result = drEvent.getResult();
+            this.getEffects().setValue("rolled", result);
+            return result >= 4;
+        }
+        return false;
+    }
+
+    @Override
+    public String getRule(){
+        return "Whenever you roll a 4 or higher, create a 3/3 colorless Robot artifact creature token. " +
+                "If you rolled 6 or higher, instead create that token and a Treasure token.";
+    }
+}
+
+class MrHousePresidentAndCEOTokenEffect extends OneShotEffect {
+
+    public MrHousePresidentAndCEOTokenEffect() {
+        super(Outcome.Benefit);
+    }
+
+    private MrHousePresidentAndCEOTokenEffect(final MrHousePresidentAndCEOTokenEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public MrHousePresidentAndCEOTokenEffect copy() {
+        return new MrHousePresidentAndCEOTokenEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (getValue("rolled") == null) {
+            return false;
+        }
+        int amount = (Integer) getValue("rolled");
+
+        if (amount >= 6) {
+            Token treasureToken = new TreasureToken();
+            treasureToken.putOntoBattlefield(1, game, source);
```

Comment:
Must put after robot token

Reply:
Out of curiosity, why does the order matter (in terms of the card text or the game mechanics)?

Reply:
It's good code style to repeat a paper rules by same code naming/order/logic/etc. Also put to battlefield order can be important for triggers or continues effects in some use cases (not here).